### ISSUE TITLE
clang-format not working after v1.17.5

### ISSUE THREAD

**Issue by svoulaz** [microsoft/vscode-cpptools#11694]
### Environment

- OS and Version: Windows 11 23H2
- VS Code Version: 1.84.2
- C/C++ Extension Version: 1.18.4

### Bug Summary and Steps to Reproduce

Bug Summary:
Starting from 1.18.3, auto formatting on save with clang-format does not work anymore - at least on my system. Just save a any source file and a popup shows "Formatting failed. See output window for details". This is the output:

c:\Users\Voulaz\.vscode\extensions\ms-vscode.cpptools-1.18.4-win32-x64/bin/../LLVM/bin/clang-format.exe -style=file -fallback-style=LLVM --Wno-error=unknown -assume-filename=[,,,]\data.hpp

The feature works fine using extension version 1.17.5

### Configuration and Logs

```shell
No c_cpp_properties.json in workspace
No "C/C++: Log Diagnostics" available from the command palette
```

**Comment by ArmoredPony** [comment]
Same on Windows 10. 1.18.4 and 1.18.3 yield that problem but clang-format itself worked outside of extension. On 1.17.5 this problem does not appear.

**Comment by sean-mcmanus** [comment]
@svoulaz 1.19.0 will add more informative logging that explains the failure, but in the meantime, can you run the command below on the command line

c:\Users\Voulaz.vscode\extensions\ms-vscode.cpptools-1.18.4-win32-x64/bin/../LLVM/bin/clang-format.exe -style=file -fallback-style=LLVM --Wno-error=unknown -assume-filename=[,,,]\data.hpp [,,,]\data.hpp

to see what the failure error message is?

**Comment by JuanPabloEs** [comment]
Hi, in my case just as a quick note it also crashes with the new versions but in 1.17.5 it works correctly.

I have run the command of the new version 1.18.4 which has the flag `-assume-filename` with the clang-format.exe of version 1.17.5 I don't know the utility of this flag but in my case it causes the formatting to be blocked. If I run the command only idicating the filename on the 1.18.4 it seems to work correctly.

![image](https://github.com/microsoft/vscode-cpptools/assets/54740762/f2ee37a8-e267-4a6f-a3c1-0c5ad5fe7acd)

**Comment by svoulaz** [comment]
@sean-mcmanus - Aha! Newer clang-format 17 complains and stops parsing the format file in case a key is duplicated. In my case:

```
[...]\.clang-format:92:1: error: duplicated mapping key 'KeepEmptyLinesAtTheStartOfBlocks'
KeepEmptyLinesAtTheStartOfBlocks:               true
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Error reading [...]\.clang-format: invalid argument
```

Should have checked out - indeed more informative output would help.

**Comment by bobbrow** [comment]
@JuanPabloEs you need to add the filename to the end of the command or else clang-format is expecting to read from stdin. We fixed the logging in 1.19.0 to include the filename at the end of the command.
